controller/src/lib.rs

```diff
@@ -20,114 +25,204 @@ pub mod plan;
 pub mod run;
 mod scheduler;
 
-pub async fn run_scheduler(nats: TypedNats, state: StateHandle) -> NeverResult {
-    let scheduler = Scheduler::new(state.clone());
-    let mut schedule_request_sub = nats.subscribe(ScheduleRequest::subscribe_subject()).await?;
-    tracing::info!("Subscribed to spawn requests.");
-
-    while let Some(schedule_request) = schedule_request_sub.next().await {
-        tracing::info!(metadata=?schedule_request.value.metadata.clone(), "Got spawn request");
-
-        if let Some(lock) = &schedule_request.value.lock {
-            tracing::info!(lock=%lock, "Request includes lock.");
-            let locked = {
-                let state = state.state();
-                let cluster = state
-                    .cluster(&schedule_request.value.cluster)
-                    .ok_or_else(|| anyhow!("Cluster does not exist."))?;
-                cluster.locked(lock)
-            };
-            tracing::info!(lock=%lock, ?locked, "Lock checked.");
-
-            if let Some(backend) = locked {
-                tracing::info!(lock=%lock, "Lock is held.");
-
-                // Anything that fails to find the drone results in an error here, since we just
-                // checked that the lock is held which implies that the drone exists.
-
-                let (drone, bearer_token) = {
-                    let state = state.state();
-                    let cluster = state
-                        .cluster(&schedule_request.value.cluster)
-                        .ok_or_else(|| anyhow!("Expected cluster to exist."))?;
-
-                    let backend_state = cluster
-                        .backend(&backend)
-                        .ok_or_else(|| anyhow!("Lock held by a backend that doesn't exist."))?;
+async fn spawn_backend(
+    ref nats: TypedNats,
```

Comment:
I don't think `ref` is necessary here? We are cloning it when we pass it in anyway, I don't see why we need it as a reference.